### PR TITLE
fix(parser-openapi): infer entities for resources with multiple schemas

### PR DESCRIPTION
**Pull request by tomhoule**
Some resources have multiple responses associated to them, e.g. for different status codes. Before this commit, we bailed when we encountered multiple response shapes. Now we try to pick the most likely (successful) one. Alternatively, we could infer a GraphQL union of these.

closes GB-5486

# Type of change

- [ ] 💔 Breaking
- [ ] 🚀 Feature
- [x] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: engine/crates/engine/src/registry/resolvers/http/mod.rs [hunk 1]
```diff
@@ -159,6 +159,13 @@ impl HttpResolver {
 }
 
 impl ExpectedStatusCode {
+    pub fn is_success(&self) -> bool {
+        match self {
+            ExpectedStatusCode::Exact(code) => 200 <= *code && *code < 300,
+            ExpectedStatusCode::Range(code_range) => code_range.contains(&200) && code_range.end < 300,
```
Comment (tomhoule) [inline engine/crates/engine/src/registry/resolvers/http/mod.rs:165]: This feels a bit yolo.

Comment (tomhoule) [inline engine/crates/engine/src/registry/resolvers/http/mod.rs:165]: ...yolo?

File: engine/crates/parser-openapi/src/parsing/grouping.rs [hunk 2]
```diff
@@ -33,13 +33,22 @@ fn find_operation_resource_schema(ctx: &super::Context, operation_index: NodeInd
...
+        // If there is more than one schema associated to the operation, we have to rely on
+        // heuristics to pick one.
+        2.. => schema_indices.into_iter().find(|schema_index| {
+            ctx.graph
+                .edges_connecting(operation_index, *schema_index)
+                .any(|edge| match edge.weight() {
+                    // This is a/the sucess response: pick it.
+                    Edge::HasResponseType { status_code, .. } => status_code.is_success(),
+                    // ?
+                    Edge::HasField { .. } => true,
```
Comment (tomhoule) [inline engine/crates/parser-openapi/src/parsing/grouping.rs:47]: ?

Comment (tomhoule) [reply engine/crates/parser-openapi/src/parsing/grouping.rs:47]: This is very much me fumbling around in the crate, the structure is nice and clear but I'm very new to it. But I think I get what was wrong and the suggested change, I pushed a new commit, tests are still passing locally.

**Review by obmarg — APPROVED** [review]

**Pull request merged**